spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/DataTables.groovy

```diff
@@ -226,19 +230,118 @@ local | 1
     thrown(MissingPropertyException)
   }
 
-  def "cells cannot reference other cells"() {
+  def 'cells can reference previous cells'() {
+    expect:
+    [a, b, c] == [0, 1, 2]
+
+    where:
+    a | b     | c
+    0 | a + 1 | b + 1
+  }
+
+  def 'cell references are pointing to the current row'() {
+    expect:
+    b == 1 + a * 2
+    c == 3 * b
+
+    where:
+    a | b         | c
+    0 | 1 + a * 2 | 3 * b
+    1 | 1 + a * 2 | 3 * b
+    2 | 1 + a * 2 | 3 * b
+  }
+
+  def "cell references are evaluated correctly in the method's name"() {
     when:
-    runner.runFeatureBody """
-expect:
-true
+    runner.runSpecBody '''
+      @Unroll def 'a = #a, b = #b'() {
+        expect:
+        true
+        
+        where:
+        a | b
+        0 | a + 1
+      }
+    '''
 
-where:
-a | b
-1 | a
-    """
+    then:
+    1 * listener.testStarted { it.methodName == "a = 0, b = 1" }
+  }
+
+  def "cells can't reference next cells"() {
+    when:
+    runner.runFeatureBody '''
+      expect:
+      false
+      
+      where:
+      a | b
+      b | 1
+    '''
 
     then:
-    thrown(MissingPropertyException)
+    thrown Exception
+  }
+
+  def "cells can't reference themselves"() {
+    when:
+    runner.runFeatureBody '''
+      expect:
+      false
+      
+      where:
+      a | b
+      1 | b + 1
+    '''
+
+    then:
+    thrown Exception
+  }
+
+  def 'cell references are working with simple parameterization also'() {
+    expect:
+    c == 2
+
+    where:
+    a << [1, 2]
+    b << [3, 4]
+    c << [b - a, b - a]
+  }
+
+  def 'data tables can be referenced from following variables'() {
+    expect:
+    c == 3
+
+    where:
+    a | b
+    1 | 2
+    1 | a + 1
+
+    c = b + 1
+  }
+
+  @Ignore
+  def 'data table elements can reference each other'() {
+    expect:
+    runner.runFeatureBody '''
+      expect:
+      g == 12
+  
+      where:
+      a = 1
+      b = a + 1
+      
+      c << [b + 1]
```

Comment:
@robfletcher and I were discussing this PR an decided to include it in the upcoming 1.1 Release, which is planned to happen when #17 is implemented and merged (hopefully within two weeks).

I'm ok with fixing it in a separate commit, as long as it is mentioned in the documentation (which is still missing :wink: ). The documentation should include examples for what is and what is not yet possible, and what will never work (e.g., referencing values from a previous execution). Could you please provide this documentation in another PR. 

Regarding 3) yeah it's unfortunate, but we do this in our spare time and as you said in 2) it's a complicated change and it takes time and requires more mental work than a simple pr so you can't just do it on the side. Anyway thanks a lot for the PR and I'm looking forward to your next ones. And look on the bright sight, my PR #17 is from September 2013! and its been in limbo until recently ;) so two month vs. two years is not that bad.